crates/rollup-boost/src/flashblocks/service.rs

```diff
@@ -69,30 +63,28 @@ struct FlashblockBuilder {
 
 impl FlashblockBuilder {
     pub fn new() -> Self {
-        Self {
-            base: None,
-            flashblocks: Vec::new(),
-        }
+        Self::default()
     }
 
     pub fn extend(&mut self, payload: FlashblocksPayloadV1) -> Result<(), FlashblocksError> {
         tracing::debug!(message = "Extending payload", payload_id = %payload.payload_id, index = payload.index, has_base=payload.base.is_some());
 
-        // Check base payload rules
-        match (payload.index, payload.base) {
-            // First payload must have a base
-            (0, None) => return Err(FlashblocksError::MissingBasePayload),
-            (0, Some(base)) => self.base = Some(base),
-            // Subsequent payloads must have no base
-            (_, Some(_)) => return Err(FlashblocksError::UnexpectedBasePayload),
-            (_, None) => {} // Non-zero index without base is fine
-        }
-
         // Validate the index is contiguous
         if payload.index != self.flashblocks.len() as u64 {
             return Err(FlashblocksError::InvalidIndex);
         }
 
+        // Check base payload rules
+        if payload.index == 0 {
```

Comment:
I would like to add a test for this before rewriting it to make sure the migration works as expected.

Reply:
https://github.com/flashbots/rollup-boost/issues/362

Reply:
https://github.com/flashbots/rollup-boost/pull/363